### ISSUE TITLE
Add capabilities to split and impute AFQDataset

### ISSUE THREAD

**Issue by richford** [richford/AFQ-Insight#98]
Dataset splitting and imputation is common and has the potential to introduce train/test leakage if done incorrectly. It's also easier to do when all of your dataset is in memory (i.e. before being batched into tensorflow or pytorch datasets.

## Proposed solution

- [ ] Make `AFQDataset` initialization more general so that user can provide `X`, `y`, `groups`, `feature_names`, `group_names`, `subjects`, `sessions`, `classes` explicitly on init.
- [ ] create a new static method `from_files()` that takes all of the filenames that are in the current `AFQDataset` init method and returns an `AFQDataset` object with all of the data read in.
- [ ] create a new `split()` method that will apply an [sklearn splitter class](https://scikit-learn.org/stable/modules/classes.html#splitter-classes) and wrap its `split` method (which usually returns tuples of indices) by returning a tuple of `AFQDataset` objects representing the train and test splits. The user is free to recursively split the test set if they want both test and validate splits.
- [ ] create a `train_test_split()` that wraps `next(split(ShuffleSplit()))` in much the same way that `train_test_split` works in sklearn.

## Topics for discussion

We often need to impute missing bundles and have to take care to fit imputation on only the training set to avoid data leakage. It would be nice to provide the user with an option to do this during splitting. One option is to add `imputer` and `imputer_kwargs` parameters to the aforesaid `split()` method and impute within the wrapper for the splitters split method. I like this option but am open to other implementations.

Tagging @arokem to continue a discussion that we had about this in person. Note my change of heart here in making `AFQDataset` more general and reading from input files using a static method.
